### PR TITLE
Remove Cargo.lock, update dependencies, update submodule, disable build grpc server api.

### PR DESCRIPTION
**Pull request by jmjoy**

### CODE REVIEW

**Comment by wu-sheng** [comment]
> Remove Cargo.lock

Remove the lock file because?

**Comment by jmjoy** [comment]
> > Remove Cargo.lock
> 
> Remove the lock file because?

Reference: <https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html>

> If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore.

**Comment by wu-sheng** [comment]
In ASF, all dependencies must be locked. Think in this way, one library will change license from Apache 2.0 to GPL even commercial, then every released version would be compromised.

**Comment by wu-sheng** [comment]
If you are upgrading versions like this PR, binary license should be updated

**Comment by jmjoy** [comment]
1. When crate as a dependency, `Cargo.lock` is not valid, even if you add it to version management.
2. In general, changing the license should promote the large version, or minor version of version 0.x, cargo will not select the incompatible version of `Cargo.toml` when handling dependencies.

File: build.rs [hunk 1]
```diff
@@ -15,8 +15,11 @@
 //
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    tonic_build::configure().compile(
-        &["./skywalking-data-collect-protocol/language-agent/Tracing.proto"],
+    tonic_build::configure().build_server(false).compile(
+        &[
+            "./skywalking-data-collect-protocol/language-agent/Tracing.proto",
+            "./skywalking-data-collect-protocol/management/Management.proto",
```
Comment (wu-sheng) [inline build.rs:21]: Why do you need this? I think we don't call management APIs from Rust side.

**Comment by wu-sheng** [comment]
Please check CI.

File: build.rs (hunk 1 above)
Comment (jmjoy) [reply build.rs:21]: I want to call `reportInstanceProperties` to report the properties such as `os_name`, `host_name`.

Comment (wu-sheng) [reply build.rs:21]: Then add this when you submit relative codes, please.

File: e2e/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,7 @@ name = "e2e"
 version = "0.1.0"
 authors = ["Shikugawa <Shikugawa@gmail.com>"]
 edition = "2021"
+publish = false
```
Comment (wu-sheng) [inline e2e/Cargo.toml:23]: What is this about?

Comment (jmjoy) [reply e2e/Cargo.toml:23]: Avoid publish `e2e` to crates.io.

**Comment by wu-sheng** [comment]
The agent test is still failure.

File: build.rs (hunk 1 above)
Comment (wu-sheng) [reply build.rs:21]: @jmjoy Could you revert this in this PR?

Comment (jmjoy) [reply build.rs]: Done.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**